aptos-move/e2e-move-tests/src/tests/fee_payer.rs

```diff
@@ -1,296 +1,106 @@
-    let carol_end = read_coin(&h, carol.address());
-    let david_end = read_coin(&h, david.address());
-    let payer_end = read_coin(&h, payer.address());
-
-    // Make sure alice pays gas
-    assert_eq!(alice_start - amount_alice - output.gas_used(), alice_end);
-    assert_eq!(bob_start - amount_bob, bob_end);
-    assert_eq!(carol_start + amount_carol, carol_end);
-    assert_eq!(david_start + amount_david, david_end);
-    // Make sure payer doesn't pays
-    assert_eq!(payer_start, payer_end);
-}
-
-#[test]
-fn test_two_to_two_transfer_fee_payer_without_feature() {
-    let mut h = MoveHarness::new_with_features(vec![], vec![FeatureFlag::GAS_PAYER_ENABLED]);
-
-    let alice = h.new_account_at(AccountAddress::from_hex_literal("0xa11ce").unwrap());
-    let bob = h.new_account_at(AccountAddress::from_hex_literal("0xb0b").unwrap());
-    let carol = h.new_account_at(AccountAddress::from_hex_literal("0xca501").unwrap());
-    let david = h.new_account_at(AccountAddress::from_hex_literal("0xda51d").unwrap());
-    let payer = h.new_account_at(AccountAddress::from_hex_literal("0xea51d").unwrap());
-
-    let amount_alice = 100;
-    let amount_bob = 200;
-    let amount_carol = 50;
-
-    let build_options = aptos_framework::BuildOptions {
-        with_srcs: false,
-        with_abis: false,
-        with_source_maps: false,
-        with_error_map: false,
-        ..aptos_framework::BuildOptions::default()
-    };
+    let output = h.run_raw(transaction);
+    assert_success!(*output.status());
 
-    let package = aptos_framework::BuiltPackage::build(
-        common::test_dir_path("../../../move-examples/scripts/two_by_two_transfer"),
-        build_options,
-    )
-    .expect("building package must succeed");
+    let alice_after = h.read_aptos_balance(alice.address());
+    let bob_after = h.read_aptos_balance(bob.address());
 
-    let code = package.extract_script_code()[0].clone();
-    let script = Script::new(code, vec![], vec![
-        TransactionArgument::U64(amount_alice),
-        TransactionArgument::U64(amount_bob),
-        TransactionArgument::Address(*carol.address()),
-        TransactionArgument::Address(*david.address()),
-        TransactionArgument::U64(amount_carol),
-    ]);
-
-    let transaction = TransactionBuilder::new(alice.clone())
-        .secondary_signers(vec![bob])
-        .fee_payer(payer)
-        .script(script)
-        .sequence_number(h.sequence_number(alice.address()))
-        .max_gas_amount(1_000_000)
-        .gas_unit_price(1)
-        .sign_fee_payer();
-
-    let output = h.executor.execute_transaction(transaction);
-    assert!(transaction_status_eq(
-        output.status(),
-        &TransactionStatus::Discard(StatusCode::UNEXPECTED_ERROR_FROM_KNOWN_MOVE_FUNCTION)
-    ));
+    assert_eq!(alice_start, alice_after);
+    assert!(bob_start > bob_after);
 }
 
 #[test]
-fn test_normal_tx_with_signer_with_fee_payer() {
+fn test_account_not_exist_with_fee_payer_create_account() {
```

Comment:
I wonder if we should do this test in api instead to make sure everything works well e2e

Reply:
From my understanding, this is comprehensive. But I have another AIP coming, so I'll make a new end to end test for another AIP in this space.

Reply:
fyi, tested in the next pr